mage_ai/orchestration/db/models/schedules.py

```diff
@@ -308,6 +309,38 @@ def in_progress_runs(
             PipelineRun.passed_sla.is_(False),
         ).all()
 
+    @safe_db_query
+    def retry_pipeline_run(
+        self,
+        pipeline_run: dict,
```

Comment:
Let's use `Dict` type here, which is imported by `from typing import Dict`
I remember `dict` type is only supported in python 3.9 and newer versions.